src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -213,21 +209,29 @@ public ModelAndView upliftRegister(@ModelAttribute("registerUserCommand") @Valid
                     return new ModelAndView(REDIRECT_RESET_INACTIVE_USER, model);
                 }
 
-                msg += "PIN user no longer valid";
+                msg = "PIN user no longer valid";
             }
 
-            if (ex.getStatusCode().equals(HttpStatus.CONFLICT)) {
-                msg = "Your account is already active.";
+            if (!ex.getStatusCode().equals(HttpStatus.CONFLICT)) {
+                ErrorHelper.showLoginError("Sorry, there was an error",
+                    String.format("Please try your action again. %s", msg),
+                    request.getRedirect_uri(),
+                    model);
+                // We use spring:hasBindErrors so make sure the 'showLoginError' is rendered to the page
+                // by adding a binding error
+                bindingResult.reject("non-existent-error-code");
+
+                return new ModelAndView(UPLIFT_REGISTER_VIEW, model);
             }
+        }
 
-            ErrorHelper.showLoginError("Sorry, there was an error",
-                msg, request.getRedirect_uri(), model);
-            // We use spring:hasBindErrors so make sure the 'showLoginError' is rendered to the page
-            // by adding a binding error
-            bindingResult.reject("non-existent-error-code");
+        model.put(EMAIL, request.getUsername());
+        model.put(REDIRECTURI, request.getRedirect_uri());
+        model.put(CLIENTID, request.getClient_id());
+        model.put(JWT, request.getJwt());
+        model.put(STATE, request.getState());
```

Comment:
Can we take the opportunity to add `NONCE` to the list of model attributes?

Reply:
I've added it to all redirects and requests now, waiting on functionals